pool.go

```diff
@@ -187,7 +188,8 @@ func (p *Pool) recreateConnection(closedConn *Connection) {
 
 		p.handleError(fmt.Errorf("failed to reconnect to %s: %w", conn.addr, err))
 		select {
-		case <-time.After(p.Opts.ReconnectWait):
+		case <-time.After(reconnectTime):
+			reconnectTime *= 2
```

Comment:
Can we have an upper bound here? We don't want this to forever increase.

Reply:
Do we want to add capped duration or retry counts limit ?